narwhals/translate.py

```diff
@@ -776,8 +778,43 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
         return decorator(func)
 
 
+def to_py_scalar(scalar_like: Any) -> Any:
+    """If a scalar is not Python native, tries to convert it to Python native.
+
+    Examples:
+        >>> import narwhals as nw
+        >>> import pandas as pd
+        >>> df = nw.from_native(pd.DataFrame({"a": [1, 2, 3]}))
+        >>> nw.to_py_scalar(df["a"].item(0))
+        1
+        >>> import pyarrow as pa
+        >>> df = nw.from_native(pa.table({"a": [1, 2, 3]}))
+        >>> nw.to_py_scalar(df["a"].item(0))
+        1
+        >>> nw.to_py_scalar(1)
+        1
+    """
+
+    pa = get_pyarrow()
+    if pa and isinstance(scalar_like, pa.Scalar):
+        return scalar_like.as_py()
+
+    cupy = get_cupy()
+    if (  # pragma: no cover
+        cupy and isinstance(scalar_like, cupy.ndarray) and scalar_like.size == 1
+    ):
+        return scalar_like.item()
+
+    np = get_numpy()
+    if np and np.isscalar(scalar_like) and hasattr(scalar_like, "item"):
+        return scalar_like.item()
+
+    return scalar_like
```

Comment:
Ok this may be _way_ too defensive programming but I am putting the question out there: should we raise for any other type which is not builtin?

```suggestion
    if scalar_like.__class__.__module__ == "builtins":
        return scalar_like
    
    msg = (
        f"Expected object convertible to a scalar, found {type(scalar_like)}. "
        "Please report a bug to https://github.com/narwhals-dev/narwhals/issues"
    )
    raise ValueError(msg)
```

Reply:
I think you are right! If we don't raise an error we should have called it `maybe_py_scalar`. Added the `ValueError` but restricting on types. Now for example dicts and lists will raise an error. What do you think?

Reply:
I think that's ok for the vast majority of use cases 👌